sdk/azidentity/aad_identity_client.go

```diff
@@ -61,7 +63,37 @@ func newAADIdentityClient(options *TokenCredentialOptions) *aadIdentityClient {
 // clientID: The client (application) ID of the service principal
 // clientSecret: A client secret that was generated for the App Registration used to authenticate the client
 // scopes: The scopes required for the token
-func (c *aadIdentityClient) authenticate(ctx context.Context, tenantID string, clientID string, clientSecret string, scopes []string) (*azcore.AccessToken, error) {
+func (c *aadIdentityClient) refreshAccessToken(ctx context.Context, tenantID string, clientID string, clientSecret string, refreshToken string, scopes []string) (*tokenResponse, error) {
+	msg, err := c.createRefreshTokenRequest(tenantID, clientID, clientSecret, refreshToken, scopes)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.pipeline.Do(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// This should never happen under normal conditions
+	if resp == nil {
+		return nil, &AuthenticationFailedError{msg: "Something unexpected happened with the request and received a nil response"}
+	}
+
+	if resp.HasStatusCode(successStatusCodes[:]...) {
+		return c.createAccessToken(resp)
+	}
+
+	return nil, &AuthenticationFailedError{inner: newAADAuthenticationFailedError(resp)}
```

Comment:
I don't think we should wrap the error here.  Instead, put the authentication failure info directly in the `AuthenticationFailedError`.

Reply:
Jeff says: "AuthenticationFailedError is authentication mechanism agnostic and should therefore not have fields that are specific to AAD. The AADAuthenticationFailedError has those fields and then is wrapped with AuthenticationFailedError to indicate the the authentication failed. If recovery code wants to find out why the authentication failed it would have to look specifically for an AAD failure and if found, then the fields correspond to that."

Reply:
Agreed that AAD-specific fields shouldn't be in AuthenticationFailedError.  However with the code the way it is at present, if this API call fails and you `panic(err)` or `fmt.Printf("%v", err)` you get zero information (it literally prints an empty string) because the message is empty, so the failure is not actionable.  At the very least we should have `Error()` include the error message from `inner.Error()` to avoid this.
If the inner error will contain more useful info beyond a message/description then agreed that keeping them distinct is valuable.  I just want to make sure we aren't adding a needless level of indirection.

Reply:
I agree I'll add the inner error description to the message

Reply:
Added the inner error description to AuthenticationFailedError: 
https://github.com/Azure/azure-sdk-for-go/pull/6560/files#diff-d1256cf2ef15d119cac7ccfaf6931cc9R79-R81